packages/twenty-server/src/workspace/messaging/services/create-companies-and-contacts/create-companies-and-contacts.service.ts

```diff
@@ -59,18 +63,24 @@ export class CreateCompaniesAndContactsService {
     const filteredParticipants = uniqueParticipants.filter(
       (participant) =>
         !alreadyCreatedContactEmails.includes(participant.handle) &&
-        participant.handle.includes('@'),
+        participant.handle.includes('@') &&
+        (isContactAutoCreationForNonWorkEmailsEnabled ||
+          isWorkEmail(participant.handle)),
     );
 
     const filteredParticipantsWithCompanyDomainNames =
       filteredParticipants?.map((participant) => ({
         handle: participant.handle,
         displayName: participant.displayName,
-        companyDomainName: getDomainNameFromHandle(participant.handle),
+        companyDomainName: isWorkEmail(participant.handle)
+          ? getDomainNameFromHandle(participant.handle)
+          : undefined,
       }));
 
-    const domainNamesToCreate = filteredParticipantsWithCompanyDomainNames.map(
-      (participant) => participant.companyDomainName,
+    const domainNamesToCreate = _.compact(
```

Comment:
You import like this 
`import compact from 'lodash/compact';` to not import the whole lodash lib